<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chao Dai" />

<meta name="date" content="2021-09-10" />

<title>Q5: Colocalization</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">prelim-prep</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Q5: Colocalization</h1>
<h4 class="author">Chao Dai</h4>
<h4 class="date">2021-09-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>prelim-prep/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdaiucprelimpreptree1cc0f3fd56d8c501d3cd53f2e50a0a3c5845a35ctargetblank1cc0f3fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/daiuc/prelim-prep/tree/1cc0f3fd56d8c501d3cd53f2e50a0a3c5845a35c" target="_blank">1cc0f3f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomdaiucprelimpreptree1cc0f3fd56d8c501d3cd53f2e50a0a3c5845a35ctargetblank1cc0f3fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/daiuc/prelim-prep/tree/1cc0f3fd56d8c501d3cd53f2e50a0a3c5845a35c" target="_blank">1cc0f3f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/q5.nb.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/q5.Rmd</code>) and HTML (<code>docs/q5.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/1cc0f3fd56d8c501d3cd53f2e50a0a3c5845a35c/analysis/q5.Rmd" target="_blank">1cc0f3f</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-02
</td>
<td>
wflow_publish(“analysis/q5.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/3342fe187471198793dfe28711c54096b1b0cd9f/docs/q5.html" target="_blank">3342fe1</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/96d80b72a27f81af3e51d3728fe31d6ae25370f9/analysis/q5.Rmd" target="_blank">96d80b7</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-02
</td>
<td>
wflow_publish(“analysis/q5.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/4823d845e2c2a745f7e311d392213911b54c1971/docs/q5.html" target="_blank">4823d84</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/56ded550fe47e438ab9997d007565fc1a9864599/analysis/q5.Rmd" target="_blank">56ded55</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-01
</td>
<td>
wflow_publish(“analysis/q5.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/ba39746529a125cd5458063e1557a119f7f9b774/docs/q5.html" target="_blank">ba39746</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/807eeb0f47f3fe55cd02229837fbaf6865987db5/analysis/q5.Rmd" target="_blank">807eeb0</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-01
</td>
<td>
wflow_publish(“analysis/q5.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/b86b160e226f810f1a6c1a88e91bbaf0edbe8dd0/docs/q5.html" target="_blank">b86b160</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/09e3d72fd3ac90c6f847a9ad5fda1cb944352409/analysis/q5.Rmd" target="_blank">09e3d72</a>
</td>
<td>
daiuc
</td>
<td>
2021-09-01
</td>
<td>
wflow_publish(“analysis/q5.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/70e791f078aa9ec5d5d2714b7923070ef2e839d7/docs/q5.html" target="_blank">70e791f</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/bc74cd87a5157abad3f4df58b84d663259139931/analysis/q5.Rmd" target="_blank">bc74cd8</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
wflow_publish(c(“analysis/q2.Rmd”, “analysis/q5.Rmd”, “analysis/q8.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/a8b7656f95041e314d661a8dfb1cad990ae6620b/docs/q5.html" target="_blank">a8b7656</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/1c835e59dbe0cc16c8c1ed29620ed83866ba2743/analysis/q5.Rmd" target="_blank">1c835e5</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
wflow_publish(c(“analysis/q2.Rmd”, “analysis/q5.Rmd”, “analysis/q8.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/533524d4dd1b98899a43b2fc69a4513630c5e88a/docs/q5.html" target="_blank">533524d</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/88e766e7c98d19a9347b2dc21b783279a5b62c6a/analysis/q5.Rmd" target="_blank">88e766e</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
wflow_publish(c(“analysis/q2.Rmd”, “analysis/q5.Rmd”, “analysis/q8.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/24d6fa97294fc9d2afb31dde88ee93e5bf356722/docs/q5.html" target="_blank">24d6fa9</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/d682820bfed599e7194cde06ade249cce2659142/docs/q5.html" target="_blank">d682820</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/84c50ffecf9be96b94748514b2cd70d88f5f5806/analysis/q5.Rmd" target="_blank">84c50ff</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
wflow_publish(“analysis/q5.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/prelim-prep/9340db6ac8b15d5a749726235213767b51aaa79e/docs/q5.html" target="_blank">9340db6</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/prelim-prep/blob/b4e3ccc34fb810c7e3a842fe8be60fce17535353/analysis/q5.Rmd" target="_blank">b4e3ccc</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-31
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/q2.Rmd”, “analysis/q5.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="a" class="section level2">
<h2>a)</h2>
<p><strong>Suppose a GWAS variant associated with the disease trait of interest is also associated with the expression of a gene X in some tissue. Can we conclude that X is a causal gene of the disease?</strong></p>
<p>No. We cannot conclude X is a causal gene ofo the disease simply because the variant is associated both with the disease trait and gene X expression.</p>
<p>Reasons include:</p>
<ul>
<li><p>The same variant associated with both disease and gene X may not be causal alleles to eiher the either the disease or gene X.</p>
<ul>
<li><p>There may be two different causal variants, and the observered variant is simply in LD with both. Thus there is no causal connection between gene X and disease trait.</p></li>
<li><p>Or any other set configurations where disease trait and eQTL do not share a causal variant as outlined in the coloc paper.</p></li>
</ul></li>
<li><p>Corregulation. The same observed variant may be causal to disease trait, gene X expression, and gene Y expression. But only Gene Y is causal of the disease trait.</p></li>
<li><p>At best eQTL explains about 2% (check GTEx paper) heritability. Thus even if causal signal do overlap, there may be other downstream mechanisms that contribute to disease trait, e.g. splicing, RNA degradation, translation, etc.</p></li>
</ul>
<hr />
</div>
<div id="b" class="section level2">
<h2>b)</h2>
<p><strong>A strategy to better integrate eQTL and GWAS data is colocalization analysis. Read the paper presenting the method <code>coloc</code> . Given association data of two traits in one region, <code>coloc</code> tests if there is a common causal variant in the region for both traits. State the five hypothesis <code>coloc</code> is evaluating. Suppose we are given the prior probability a SNP is associated with trait 1, <span class="math inline">\(p_1\)</span> , with trait 2, <span class="math inline">\(p_2\)</span> , and with both traits <span class="math inline">\(p_{12}\)</span> . Let <span class="math inline">\(S\)</span> be a configuration: it is a pair of binary vector of whether a SNP is causal to trait 1 and trait 2 (see Figure 1 for example). What is the prior probability of <span class="math inline">\(S\)</span> when <span class="math inline">\(S\)</span> belongs to each of the five hypothesis?</strong></p>
<p>Five hypothesis:</p>
<ul>
<li><span class="math inline">\(H_0\)</span>: No causal association with either trait 1 or trait 2
<ul>
<li><span class="math inline">\(P(S_0) = p_0^Q\)</span></li>
</ul></li>
<li><span class="math inline">\(H_1\)</span>: 1 causal SNP associated with trait 1, but no causal SNP for trait 2
<ul>
<li><span class="math inline">\(P(S_1) = p_0^{Q-1}p_1\)</span></li>
</ul></li>
<li><span class="math inline">\(H_2\)</span>: 1 causal SNP associated with trait 2, but no causal SNP for trait 1
<ul>
<li><span class="math inline">\(P(S_2) = p_0^{Q-1}p_2\)</span></li>
</ul></li>
<li><span class="math inline">\(H_3\)</span>: 1 causal SNP associated with trait 1, another different SNP associated with trait 2
<ul>
<li><span class="math inline">\(P(S_3) = p_0^{Q-2} p_1 p_2\)</span></li>
</ul></li>
<li><span class="math inline">\(H_4\)</span>: same SNP causal to both trait 1 and trait 2
<ul>
<li><span class="math inline">\(P(S_4) = p_0^{Q-1}p_{12}\)</span></li>
</ul></li>
</ul>
<hr />
</div>
<div id="c" class="section level2">
<h2>c)</h2>
<p><strong>The evidence of SNP association of a trait can be expressed as Bayes factor (BF), which compares two models, the SNP is associated with the trait vs. not associated. Suppose we know BF of each SNP in the region with respect to the two traits. Derive the posterior probability of colocalization in terms of the BFs of SNPs.</strong></p>
<p>Assuming <span class="math inline">\(p_0 \approx 1\)</span>:</p>
<p><span class="math display">\[
\begin{align*}
{\bf{H_0:}} \hspace{1cm} \frac{P(H_0 | D)}{P(H_0 | D)} &amp;= 1  \tag 1 \\
{\bf{H_1:}} \hspace{1cm} \frac{P(H_1 | D)}{P(H_0 | D)} &amp; = \sum_{S \in H_1} \frac{P(D | S)}{P(D | S_0)} \frac{P(S)}{P(S_0)} \approx p_1 \sum_{j=1}^{Q}BF_j^1 \tag 2 \\
{\bf{H_2:}} \hspace{1cm} \frac{P(H_2 | D)}{P(H_0 | D)} &amp; = \sum_{S \in H_2} \frac{P(D | S)}{P(D | S_0)} \frac{P(S)}{P(S_0)} \approx p_2 \sum_{j=1}^{Q}BF_j^2 \tag 3 \\
{\bf{H_3:}} \hspace{1cm}\frac{P(H_3 | D)}{P(H_0 | D)} &amp;= \sum_{S \in H_3} \frac{P(D | S)}{P(D | S_0)} \frac{P(S)}{P(S_0)} \approx p_1 p_2  \sum_{j, k, j\ne k}BF_j^1 BF_k^2 \tag 4 \\
{\bf{H_4:}} \hspace{1cm} \frac{P(H_4 | D)}{P(H_0 | D)} &amp;= \sum_{S \in H_4} \frac{P(D | S)}{P(D | S_0)} \frac{P(S)}{P(S_0)} \approx p_2 \sum_{j=1}^{Q}BF_j^1 BF_j^2 \tag 5 \\
P(H_0 | D) &amp;+ P(H_1 | D) + P(H_2 | D) + P(H_3 | D) + P(H_4 | D) = 1 \tag 6
\end{align*}
\]</span></p>
</div>
<div id="solving-systems-of-equations-above-2---6-we-can-solve-for-ph_0-d-ph_1-d-ph_2-d-ph_3-d-and-ph_4-d." class="section level2">
<h2>Solving systems of equations above (2 - 6), we can solve for <span class="math inline">\(P(H_0 | D), P(H_1 | D) , P(H_2 | D) , P(H_3 | D)\)</span>, and <span class="math inline">\(P(H_4 | D)\)</span>.</h2>
</div>
<div id="d" class="section level2">
<h2>d)</h2>
<p><strong>It was found that results of <code>coloc</code> are sensitive to prior parameters. A program <code>Enloc</code> is designed to address this problem. Explain its main ideas.</strong></p>
<p>In stead of predefining priors such as <span class="math inline">\(p_1\)</span>, <span class="math inline">\(p_2\)</span>, <span class="math inline">\(p_{12}\)</span> in <code>coloc</code>, <code>Enloc</code> equivalently estimates the priors, in the format of enrichment parameters <span class="math inline">\(\alpha_0\)</span>.</p>
<p>Specifically, it first model GWAS as:</p>
<p><span class="math display">\[
{\bf {y}} = \sum_{i=1}^p \beta_i {\bf{g}}_i + {\bf{e}}, {\bf{e}} \sim N(0, \tau^{-1}I).
\]</span> It then integrate molecular QTL data as annotation <span class="math inline">\(\bf{d}\)</span>, where <span class="math inline">\(\bf{d}\)</span> is a binary indicator variable. To model the uncertainties, <code>Enloc</code> further modeled <span class="math inline">\(\bf{d}\)</span> as a latent variable following the distribution: <span class="math display">\[
{\bf{d}} \sim Pr({\bf{d}} | {\bf{Y}}_{qtl}, {\bf{G}}_{qtl} ).
\]</span></p>
<p>Thus, the colocalization probability can be written as: <span class="math display">\[
Pr(\gamma_i = 1, d_i = 1 | {\bf{y}},{\bf{G}}, {\bf{Y}}_{qtl}, {\bf{G}}_{qtl}).
\]</span></p>
<p><code>Enloc</code> takes 3 primary steps to perform the inference:</p>
<ol style="list-style-type: decimal">
<li>Estimate the enrichment parameters <span class="math inline">\({\bf{\alpha}}\)</span></li>
<li>Taking estimated <span class="math inline">\({\bf{\alpha}}\)</span>, perform fine mapping using approximated PIP <span class="math inline">\(Pr({\bf{\gamma}} | {\bf{y}},{\bf{G}}, {\bf{Y}}_{qtl}, {\bf{G}}_{qtl}, {\bf{\alpha}})\)</span>.</li>
<li>Evaluate colocalization by computing SCP <span class="math inline">\(Pr(\gamma_i = 1, d_i = 1 | {\bf{y}},{\bf{G}}, {\bf{Y}}_{qtl}, {\bf{G}}_{qtl})\)</span>, and RCP, which is the sum of SCPs.</li>
</ol>
<p><em>NOTE: draw a detailed flowchart to demonstrate all the steps taken</em></p>
<hr />
</div>
<div id="e" class="section level2">
<h2>e)</h2>
<p><strong>A naive approach to colocalization is: perform fine-mapping of both traits separately, and this allows one to assess how often a SNP is a causal variant to both traits. It would be the product of Posterior Inclusion Probabilities (PIPs) of the SNP with respect to the two traits. Explain the shortcoming of this approach, in light of <code>Enloc</code>.</strong></p>
<p>Performing fine-mapping separately effectively assumes that the associations to both traits are mututally independent from one another. In the framework of <code>Enloc</code>, it means <span class="math inline">\(\alpha_1 = 0\)</span>, and equivalently, the colocalization probability can be rewritten as: <span class="math display">\[
Pr(\gamma_i = 1, d_i = 1 | {\bf{y}},{\bf{G}}, {\bf{Y}}_{qtl}, {\bf{G}}_{qtl}) \
 = Pr(\gamma_i = 1| {\bf{y}},{\bf{G}}) \cdot Pr( d_i = 1 | {\bf{Y}}_{qtl}, {\bf{G}}_{qtl}).
\]</span></p>
<p>Clearly, this is represents a special case of the model in <code>Enloc</code>. One can expect that by defining independence a priori, the number of colocalized signals is likely underestimated. Crucially, one important motivation of performing colocalization is to discover causal relationships between a molecular QTL and a complex trait. Thus assuming independence greatly undermine the purpose of the analyses.</p>
<hr />
</div>
<div id="f" class="section level2">
<h2>f)</h2>
<p><strong>The key parameter of <code>Enloc</code> is called <span class="math inline">\(\alpha_1\)</span>. Define the parameter, and explain why it is important in your own language.</strong></p>
<p><span class="math display">\[
log \left[ \frac{Pr(\gamma_i = 1)}{Pr(\gamma_i = 0)} \right] = \alpha_0 + \alpha_1 d_i \\
Pr(\gamma_i = 1, d_i = 1) = \frac{p_\gamma}{1+\frac{1-p_d}{p_d}e^{-\alpha_1}} \\
\rho = Pr(d_i = 1 | \gamma_i = 1) = \frac{1}{1+\frac{1-p_d}{p_d}e^{-\alpha_1}}
\]</span></p>
<hr />
</div>
<div id="g" class="section level2">
<h2>g)</h2>
<p><strong>Explain how <code>Enloc</code> is related to <code>coloc</code>, and why it may provide some advantages.</strong></p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
