---
title: "Q5: Colocalization"
author: "Chao Dai"
date: "2021-09-10"
output: html_notebook
---


#### a) Suppose a GWAS variant associated with the disease trait of interest is also associated with the expression of a gene X in some tissue. Can we conclude that X is a causal gene of the disease?

No. We cannot conclude X is a causal gene ofo the disease simply because the variant is associated both with the disease trait and gene X expression.

Reasons include:

* The same variant associated with both disease and gene X may not be causal alleles to eiher the either the disease or gene X.

    * There may be two different causal variants, and the observered variant is simply in LD with both. Thus there is no causal connection between gene X and disease trait. 

    * Or any other set configurations where disease trait and eQTL do not share a causal variant as outlined in the coloc paper. 

* Corregulation. The same observed variant may be causal to disease trait, gene X expression, and gene Y expression. But only Gene Y is causal of the disease trait. 

* At best eQTL explains about 2% (check GTEx paper) heritability. Thus even if causal signal do overlap, there may be other downstream mechanisms that contribute to disease trait, e.g. splicing, RNA degradation, translation, etc.

------------------------------------------------------------------------

#### b) A strategy to better integrate eQTL and GWAS data is colocalization analysis. Read the paper presenting the method coloc1 . Given association data of two traits in one region, coloc tests if there is a common causal variant in the region for both traits. State the five hypothesis `coloc` is evaluating. Suppose we are given the prior probability a SNP is associated with trait 1, p1 , with trait 2, p2 , and with both traits p12 . Let S be a configuration: it is a pair of binary vector of whether a SNP is causal to trait 1 and trait 2 (see Figure 1 for example). What is the prior probability of S when S belongs to each of the five hypothesis?

Five hypothesis: 

* $H_0$: No causal association with either trait 1 or trait 2 
* $H_1$: 1 causal SNP associated with trait 1, but no causal SNP for trait 2 
* $H_2$: 1 causal SNP associated with trait 2, but no causal SNP for trait 1 
* $H_3$: 1 causal SNP associated with trait 1, another different SNP associated with trait 2 
* $H_4$: same SNP causal to both trait 1 and trait 2

Prior probability of S:

------------------------------------------------------------------------

#### c) The evidence of SNP association of a trait can be expressed as Bayes factor (BF), which compares two models, the SNP is associated with the trait vs. not associated. Suppose we know BF of each SNP in the region with respect to the two traits. Derive the posterior probability of colocalization in terms of the BFs of SNPs.

------------------------------------------------------------------------

#### d) It was found that results of coloc are sensitive to prior parameters. A program Enloc is designed to address this problem2 . Explain its main ideas.

------------------------------------------------------------------------

#### e) A naive approach to colocalization is: perform fine-mapping of both traits separately, and this allows one to assess how often a SNP is a causal variant to both traits. It would be the product of Posterior Inclusion Probabilities (PIPs) of the SNP with respect to the two traits. Explain the shortcoming of this approach, in light of Enloc.

------------------------------------------------------------------------

#### f) The key parameter of Enloc is called Î±1. Define the parameter, and explain why it is important in your own language.

------------------------------------------------------------------------

#### g) Explain how Enloc is related to coloc, and why it may provide some advantages.

